### PR TITLE
Feature/en 558 tab set component

### PR DESCRIPTION
**Pull request by m7kvqbe1**
## Related Jira issue:

https://rn-nelson.atlassian.net/browse/EN-558

## Overview

Implement the base tab set component within the react-component-library.

## Reason

>We have a need for tabsets in the docsite. These should follow the same default stying as the component library.

## Work carried out

- [x] Implement tab set component
- [x] Add automated tests
- [x] Perform UAT down to Chrome 38 (and various form factors)

## Screenshot

Screenshot 2019-06-17 at 12 47 57
Screenshot 2019-06-17 at 13 38 50

## Developer notes

This initial implementation meets the docs-site requirements but will need to be expanded upon as component library requirements evolve (there is still an open design ticket at this point in time).

### CODE REVIEW

**Comment by antictor** [comment]
How does the focus/active state compare to button and input?

File: packages/css-framework/src/components/_tab-set.scss [hunk 1]
```diff
@@ -0,0 +1,49 @@
+$active-border-color: #2a77C7;
+
+.rn-tab-set {
+  display: flex;
+  flex-flow: column;
+}
+
+.rn-tab-set__head {
+  border-bottom: 1px solid #e2e9ee;
```
Comment (antictor) [inline packages/css-framework/src/components/_tab-set.scss]: is this in the colour palette?

File: packages/css-framework/src/components/_tab-set.scss [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    border-color: $active-border-color;
+  }
+}
+
+.rn-tab-set__tab {
+  margin: 0 spacing(1);
+  padding: spacing(4) spacing(8);
+  border: none;
+  font-size: font-size(xs);
+  font-weight: 700;
```
Comment (antictor) [inline packages/css-framework/src/components/_tab-set.scss:35]: Starting to think we need vars for weight

**Comment by m7kvqbe1** [comment]
> How does the focus/active state compare to button and input?

I'll update to match the button component.

File: packages/react-component-library/src/components/TabSet/index.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+              return (
+                <li
+                  key={uuid()}
+                  className={`rn-tab-set__tab-item ${
+                    index === activeTab ? 'is-active' : ''
+                  }`}
+                  data-testid="tab"
+                >
+                  <button
```
Comment (antictor) [inline packages/react-component-library/src/components/TabSet/index.tsx:41]: so.. when you click on a tab it's a button that marks itself as active?

Where's the link... where's the thing that says when I click on a tab, do the thing that tab has been told to do?

Comment (m7kvqbe1) [reply packages/react-component-library/src/components/TabSet/index.tsx:41]: When you click on a tab it sets some internal state (`activeTab`) to the index of the tab that was clicked. An `is-active` CSS class is then conditionally applied to the appropriate content pane based on this state.

Are you suggesting an expansion of this functionality in order to add custom actions when tabs are clicked? Invoking a custom callback or something.

File: packages/css-framework/src/components/_tab-set.scss (hunk 2 above)
Comment (m7kvqbe1) [reply packages/css-framework/src/components/_tab-set.scss:35]: Sounds good to me - here i've used the value that most closely reflected the screenshots in the Jira ticket.

Comment (m7kvqbe1) [reply packages/css-framework/src/components/_tab-set.scss:35]: Lets discuss this on Wednesday - Designers have already mentioned the need to do a proper review the CSS framework.

File: packages/react-component-library/src/components/TabSet/index.tsx (hunk 3 above)
Comment (antictor) [reply packages/react-component-library/src/components/TabSet/index.tsx:41]: I thought that was the point of the tab, not only that it updates itself but it calls a callback to notify that it's state has changes, so you'd use it to show the correct chunk of content in a page

Comment (m7kvqbe1) [reply packages/react-component-library/src/components/TabSet/index.tsx:41]: The current implementation isn't just the tabs, it also includes hidden panes for the content. I could decouple the tabs from the showing / hiding of the content panes and just make clicking the tabs invoke a supplied callback instead.

Comment (m7kvqbe1) [reply packages/react-component-library/src/components/TabSet/index.tsx:41]: This functionality has now been implemented.

**Review by 09txzS1qs — COMMENTED** [review]
Is there still work to be done on this? Will approve once @hxltrhuxze has review too.

**Comment by m7kvqbe1** [comment]
> Is there still work to be done on this? Will approve once @hxltrhuxze has review too.

This is now complete and is ready to be reviewed / merged.

**Review by antictor — APPROVED** [review]

**Review by hxltrhuxze — APPROVED** [review]

**Pull request merged**